File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.ts
function countKDifference(nums: number[], k: number): number {
  const indices = {}
  let count = 0
  for (let num of nums) {
    indices[num] = indices[num] || 0
    count += indices[num]
    indices[num + k] = indices[num + k] || 0
    indices[num + k]++
    indices[num - k] = indices[num - k] || 0
    indices[num - k]++
  }
  return count
}

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def count_k_difference(nums, k)
  indices = Hash.new(0)
  count = 0
  nums.each do |num|
    count += indices[num]
    indices[num+k] += 1
    indices[num-k] += 1
  end
  return count
end